src/cal/types.rs

```diff
@@ -0,0 +1,62 @@
+use chrono::format::ParseResult;
+use chrono::offset::Utc;
+use chrono::DateTime;
+use chrono::Duration;
+
+use std::cmp::Ordering;
+
+pub struct Event {
+    organizer: String,
+    description: String,
+    date: DateTime<Utc>,
+    duration: Duration,
+}
+
+impl Event {
+    pub fn new(
+        organizer: String,
+        description: String,
+        date: String,
```

Comment:
Just have the user pass in a parsed date. `Event` shouldn't restrict you to one date format.